src/vtp/web/api/main.py

```diff
@@ -22,19 +41,63 @@ async def root() -> dict:
     return {"version": "0.1.0"}
 
 
+# Endpoint #1
 @app.get("/vote/")
 async def get_vote_store_id() -> dict:
     """Create and store a unique Vote Store ID for each client"""
-    vote_store_id = str(random.randrange(100000, 999999))
+
+    svdo = SetupVtpDemoOperation(
+        election_data_dir=DEFAULT_ELECTION_DATA_DIR,
+        verbosity=VERBOSITY,
+    )
+    guid = svdo.run(
+        guid_client_store=True,
+    )
+
+    vote_store_id = guid
     # add VoteStoreID to list
     vote_store_ids.append(vote_store_id)
     return {"VoteStoreID": vote_store_id}
 
 
+# Endpoint #2
 @app.get("/vote/{vote_store_id}")
 async def get_empty_ballot(vote_store_id: str) -> dict:
     """Return an empty ballot for a given Vote Store ID"""
+
+    if vote_store_id in vote_store_ids:
+        # Cet a (the) blank ballot from the backend
+        cbo = CastBallotOperation(
+            guid=vote_store_id,
+            verbosity=VERBOSITY,
+        )
+        empty_ballot = cbo.run(
+            blank_ballot=DEFAULT_BLANK_BALLOT,
+            return_bb=True,
+        )
+        return {"blank-ballot": empty_ballot}
+    return {"error": "VoteStoreID not found"}
+
+
+# Endpoint #3
+@app.get("/vote/cast-ballot/{vote_store_id}")
+async def cast_ballot(vote_store_id: str, cast_ballot_json: dict = None) -> dict:
+    """
+    Uploads the ballot; the backend accepts it, merges it, and returns
```

Comment:
As noted before, move this backend logic to a separate module. The API really doesn't care what the backend does with the data, only whether it was successful or not.

If handing the completed ballot to the backend is successful, the VoteStoreID needs to be removed from the `vote_store_ids` list so the ID can't be reused to vote again.

Reply:
But the client needs the same ID to verify the ballot_check and tally the contest.  ?

Reply:
Then have a state attached to a given ID indicating whether it's been voted, verified, or tallied. The internals of that can be whatever makes sense.

Reply:
@windoverwater We could continue to store the ID on the client for as long as needed. But, I don't think the client should be able to request **another** ballot if they've already cast one with a specific ID. I think @ion-oset is proposing a more sophisticated data structure than a simple list to prevent double-voting, but I'm not sure we need to store the ID on the server after the vote is cast and recorded. Let's figure this out soon, though.

Reply:
Note - I don't have me head around the bigger issues here yet, but regarding this one snippet, for the client to verify the ballot check (to provide E2EV) the backend needs to run git commands and those commands should only be run in the specific ID (guid git workspace) that was created for that specific voter.  This is due to the way we designed the 5 endpoints (which I think of as a good thing).  Without redesigning the current endpoints, which I would prefer not to do as that may cause new backend code to be written - which is more refactoring - endpoints 4 and 5 need to specify an ID (a guid) in which to run the git commands.  Hope this helps.

And yes, I lean towards a simple design where we just do not delete the guid.  It would be good to allow the voter to verify their ballot check many times as well as run the tally many times.  I don't understand where the ID is stored on the client (as in, is it stored in 'a browser tab' or is it stored 'in a browser'), but I am ok with the voter being able to vote multiple times or not being able to vote multiple times FWIIW.